File: solutions/fortran/reverse-string/1/reverse_string.f90
module reverse_string
  implicit none
contains

  function reverse(input) result(reversed)
    character(*), intent(in) :: input
    character(len=len(input)) :: reversed
    integer :: i, t

    do i = 1, len(input)
      t = len(input) - i + 1
      reversed(i:i) = input(t:t)  
    end do
  
  end function

end module

File: solutions/free-pascal/reverse-string/1/ReverseString.pas
unit ReverseString;

{$mode ObjFPC}{$H+}

interface

function reverse(const value : string) : string;

implementation

uses SysUtils;

function reverse(const value : string) : string;
var i : integer;
begin
  result := '';
  if value='' then
    exit('');
  i := length(value);
  repeat
    result := result + value[i];
    dec(i);
  until i=0;
end;

end.
